### PR TITLE
Language Server - parseVariable initializations

### PR DESCRIPTION
**Pull request by fredg1**
manual reset of https://github.com/kolmafia/kolmafia/pull/8

### CODE REVIEW

File: src/net/sourceforge/kolmafia/textui/Parser.java
Comment (heeheehee-kolmafia) [inline src/net/sourceforge/kolmafia/textui/Parser.java:906]: Hm. If you really wanted access to the variable name, I suppose you could pass in the Variable (or VariableReference) here instead of just the Type. Doesn't make sense to initialize something other than a variable, anyhow.

**Review by heeheehee-kolmafia — APPROVED** [review]
Re-approving, since this looks like it's the same as pre-transition.

(One optional suggestion, but feel free to skip if you don't want to make that change.)

File: src/net/sourceforge/kolmafia/textui/Parser.java
Comment (fredg1) [reply src/net/sourceforge/kolmafia/textui/Parser.java:906]: I'm mostly reluctant to add a parameter for a single, edge-case purpose.
At the same time, I guess it would still make sense, since that Variable is tied to the assignment...

At at the same the same time time, one of the end-results of the next step is to add the exact location of the errors in the error message, so even without the name of the variable, people will still be able to tell which expression brought the error...

**Comment by fredg1** [comment]
Need new approval after solving a merge conflict

File: src/net/sourceforge/kolmafia/textui/Parser.java
Comment (heeheehee-kolmafia) [reply src/net/sourceforge/kolmafia/textui/Parser.java:911]: Oh, I meant in lieu of the Type, since the Variable(Reference) knows what its type is.

**Review by heeheehee-kolmafia — DISMISSED** [review]

**Comment by fredg1** [comment]
oh... yeah, that works

File: src/net/sourceforge/kolmafia/textui/Parser.java
Comment (fredg1) [reply src/net/sourceforge/kolmafia/textui/Parser.java:911]: Oh, yeah, that works

**Comment by fredg1** [comment]
Note: VariableReference.toString() => return this.target.getName();

**Review by heeheehee-kolmafia — APPROVED** [review]

**Pull request merged**
